### DISCUSSION TITLE
How about introducing Design Kit concept and standard for open source pdks such as IHP OpenPDK and skywater130?

### DISCUSSION THREAD

**Discussion by mkghub** [ra3xdh/qucs_s#849] [Ideas]
How about introducing the concept of design-kit? Then we can load and unload design-kits for each process and do the simulation conveniently? For schematic simulations, we need corresponding device symbols, spice models, and a function for netlisting. And some callback functions to calculate some parameters for each device symbol. If interface standards related this items are defined, and some functions for importing each design-kits can be added?
   Maybe we can make skywater-130 open design-kit for Qucs-s first?

**Comment by ra3xdh** [comment]
Yes, the design kits concept would be useful and technically possible to implement. Something could be implemented based on existing library approach. You may also look how Qucs-S support in IHP PDK is done: https://github.com/IHP-GmbH/IHP-Open-PDK Currently Qucs-S doesn't support callbacks on Library level. It's unclear how to better implement this. I am not planning to add such extension in the near future but certainly accept a patch implementing this function. 

At first point we can draft a plan what is needed to implement design kits support within Qucs-S. You may also look at https://github.com/Qucs/qucs/issues/659 This adds XML-defined system devices for old Qucs. May be useful for future extensions.

**Reply by mkghub** [reply]
For your reference, there is a relatively new open-source eda implemented in python, Revolution EDA 

- https://github.com/eskiyerli/revedaRelease.
- GUI is implemented with PySide6.
- Callback function is supported by python functions(Instance parameters can be also python functions that allow the dynamic calculation of parameters for each instance).
- Includes symbol library imported from IHP-Open-PDK as well as parametric layout cells for NMOS, silicide resistor and MIMCAP. Revolution EDA is not associated with IHP.

**Comment by KrzysztofHerman** [comment]
@mkghub maybe we can move the discussion to this thread #876. Although it is about IHP-Open-PDK the extensions should remain PDK agnostic.

**Reply by mkghub** [reply]
Ok for moving.
